scripts/sanitycheck

```diff
@@ -599,27 +592,31 @@ class BinaryHandler(Handler):
             proc.wait()
             self.returncode = proc.returncode
 
-        self.metrics["handler_time"] = time.time() - start_time;
+        handler_time = time.time() - start_time
 
         if options.enable_coverage:
-            returncode = subprocess.call(["GCOV_PREFIX=" + self.outdir,
-                "gcov", self.sourcedir, "-b", "-s", self.outdir], shell=True)
+            subprocess.call(["GCOV_PREFIX=" + self.build_dir,
+                "gcov", self.sourcedir, "-b", "-s", self.build_dir], shell=True)
 
         self.try_kill_process_by_pid()
 
-        # FIME: This is needed when killing the simulator, the console is
+        # FIXME: This is needed when killing the simulator, the console is
         # garbled and needs to be reset. Did not find a better way to do that.
 
         subprocess.call(["stty", "sane"])
         self.instance.results = harness.tests
-        if self.terminated==False and self.returncode != 0:
+        if not self.terminated and self.returncode != 0:
             #When a process is killed, the default handler returns 128 + SIGTERM
             #so in that case the return code itself is not meaningful
-            self.set_state("error", {})
+            self.set_state("error", handler_time)
```

Comment:
> I can see the issue, it was masked by the fact valgrind is disabled in code. Will fix the reporting and making valgrind scanning an option in a separate PR

Selecting valgrind as a cmd line option would be certainly nice, and is indeed a stretch for this PR. (The reporting in today's sanitychek when a case fails due to valgrind is just pointing to the handler.log (there was no easy way to tell which failed) but that is a relatively minor thing, as at least it reported the case as failed). 

The issue would seem to be in that this line is not having any effect in the test status anymore.

Reply:
i have a fix for all of that already and looking into pointing to the valgrind log file instead etc etc, so this will be coming.